### PR TITLE
Added interface to handler, and fixed testing

### PR DESCRIPTION
**Pull request by bokimilinkovic**

### CODE REVIEW

File: handler/dto/Book.go [hunk 1]
```diff
@@ -20,3 +27,41 @@ type author struct {
 type cover struct {
 	Url string `json:"small"`
 }
+
+
+func CreateBookModelFromBook(b Book) (bm model.Book) {
```
Comment (sfqi) [inline handler/dto/Book.go]: What does this have to do with the openlibrary client interface?

File: handler/handlers.go [hunk 2]
```diff
@@ -7,21 +7,21 @@ import (
...
 	"github.com/library/openlibrary"
 	"github.com/library/repository/mock"
+
 )
 
-var openLibraryURL = os.Getenv("LIBRARY")
-var client = *openlibrary.NewClient(openLibraryURL)
 
+//var client = *openlibrary.NewClient(OpenLibraryURL)
+type OpenLibraryClient interface {
```
Comment (sfqi) [inline handler/handlers.go]: Does this need to be exported?

File: openlibrary/client.go [hunk 3]
```diff
@@ -19,6 +19,10 @@ type Client struct {
 	url string
 }
 
+type CreateBookRequest struct {
```
Comment (sfqi) [inline openlibrary/client.go]: Why here? Are we creating a book on openlibrary.org?
Also, what does this have to do with the client interface?

File: main.go [hunk 4]
```diff
@@ -1,14 +1,21 @@
...
 	"net/http"
+	"os"
 
 	"github.com/gorilla/mux"
 	"github.com/library/handler"
 )
 
 func main() {
-	bookHandler := handler.Book{}
+	OpenLibraryUrl := os.Getenv("LIBRARY")
```
Comment (sfqi) [inline main.go]: Open -> open

**Comment by bokimilinkovic** [comment]
jel moze sad da se mrdzuje?

File: repository/mock/database.go [hunk 5]
```diff
@@ -57,3 +63,41 @@ func (bm Db) FindBookById(id int) (book model.Book, location int, found bool) {
 
 	return book, location, found
 }
+
+
+func CreateBookModelFromBook(b dto.Book) (bm model.Book) {
```
Comment (sfqi) [inline repository/mock/database.go]: Can you just put it back where it was in the first place?

Comment (bokimilinkovic) [reply repository/mock/database.go]: yes
